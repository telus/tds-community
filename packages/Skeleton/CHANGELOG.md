# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [3.1.0](https://github.com/telus/tds-community/compare/@tds/community-skeleton@3.0.3...@tds/community-skeleton@3.1.0) (2021-09-21)


### Features

* **community-skeleton:** react 17 and styled components 5 ([5717b81](https://github.com/telus/tds-community/commit/5717b81e4e5e2673755efe548f2ffe0a1629d826))





## [3.0.3](https://github.com/telus/tds-community/compare/@tds/community-skeleton@3.0.2...@tds/community-skeleton@3.0.3) (2020-09-09)


### Bug Fixes

* **community-skeleton:** update util-helpers version ([4d7f67f](https://github.com/telus/tds-community/commit/4d7f67ff65561b0972bd53259c450e8198611697))





## [3.0.2](https://github.com/telus/tds-community/compare/@tds/community-skeleton@3.0.1...@tds/community-skeleton@3.0.2) (2020-03-19)

**Note:** Version bump only for package @tds/community-skeleton





## [3.0.1](https://github.com/telus/tds-community/compare/@tds/community-skeleton@3.0.0...@tds/community-skeleton@3.0.1) (2019-09-17)

**Note:** Version bump only for package @tds/community-skeleton





# [3.0.0](https://github.com/telus/tds-community/compare/@tds/community-skeleton@2.0.3...@tds/community-skeleton@3.0.0) (2019-08-29)


### Features

* **community-skeleton:** upgrade to styled components ([a44b8ff](https://github.com/telus/tds-community/commit/a44b8ff))


### BREAKING CHANGES

* **community-skeleton:** styled components now a peer dependency





## [2.0.3](https://github.com/telus/tds-community/compare/@tds/community-skeleton@2.0.2...@tds/community-skeleton@2.0.3) (2019-06-13)

**Note:** Version bump only for package @tds/community-skeleton





## [2.0.2](https://github.com/telus/tds-community/compare/@tds/community-skeleton@2.0.1...@tds/community-skeleton@2.0.2) (2019-04-18)

**Note:** Version bump only for package @tds/community-skeleton





<a name="2.0.1"></a>
## [2.0.1](https://github.com/telus/tds-community/compare/@tds/community-skeleton@2.0.0...@tds/community-skeleton@2.0.1) (2018-10-22)




**Note:** Version bump only for package @tds/community-skeleton

<a name="2.0.0"></a>
# [2.0.0](https://github.com/telus/tds-community/compare/@tds/community-skeleton@1.0.0...@tds/community-skeleton@2.0.0) (2018-10-22)


### Bug Fixes

* **community-skeleton:** remove browser prefixes ([1a67f8d](https://github.com/telus/tds-community/commit/1a67f8d))


### Features

* **community-skeleton:** added animations, removed variants ([#115](https://github.com/telus/tds-community/issues/115)) ([e72073e](https://github.com/telus/tds-community/commit/e72073e))


### BREAKING CHANGES

* **community-skeleton:** removed property variant

omit
