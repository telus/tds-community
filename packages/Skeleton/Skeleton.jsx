import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { colorGainsboro, colorAthensGrey } from '@tds/core-colours'
import safeRest from '@tds/shared-safe-rest'

const keyframes = require('styled-components').keyframes

export const SIZES = { xs: 18, sm: 24, md: 36, lg: 48, xl: 64 }

const skeletonTextShimmer = keyframes`
  0% {
    background-position: 200% 0;
  }
  100% {
    background-position: -100% 0;
  }
`
// Not using object notation as styled-components does not currently support animations in objects
const StyledSkeleton = styled.span`
${({ safeSize }) => `
  height: ${safeSize}px;
  min-width: ${safeSize}px;
  border-radius: ${safeSize / 2}px;`}

  width: ${({ isFixedCharacterWidth, characters, safeSize }) =>
    `${isFixedCharacterWidth ? `${characters * safeSize}px` : '100%'}`};

  max-width: 100%;
  display: inline-block;
  overflow: hidden;

  background-color: ${colorGainsboro};
  background-repeat: no-repeat;
  background-position: 0 0;
  background-size: 200% 100%;
  background-image: linear-gradient(
    to right,
    ${colorGainsboro},
    ${colorAthensGrey},
    ${colorGainsboro}
  );

  animation-name: ${skeletonTextShimmer};
  animation-duration: 2s;
  animation-delay: 2s;
  animation-iteration-count: infinite;
`
/**
 * Skeleton component for mocking content while it is loading.
 *
 * @version ./package.json
 */
const Skeleton = ({ characters, size, ...rest }) => {
  const parsedCharacters = parseInt(characters, 10)
  const safeCharacters = parsedCharacters || 0
  const safeSize = SIZES[size] || SIZES.xs

  const isFixedCharacterWidth = safeCharacters > 0

  return (
    <StyledSkeleton
      {...safeRest(rest)}
      characters={characters}
      safeSize={safeSize}
      isFixedCharacterWidth={isFixedCharacterWidth}
      aria-busy="true"
    />
  )
}

Skeleton.propTypes = {
  /** Number of "M" characters to emulate (full width & height per size) <br />
   <em>NOTE: default and overflow states both assume to 100% width instead</em> */
  characters: PropTypes.number,

  /** Size of text to emulate (line height in pixels) */
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
}

Skeleton.defaultProps = {
  characters: 0,
  size: 'xs',
}

export default Skeleton
