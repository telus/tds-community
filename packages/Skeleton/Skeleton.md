### Basic Usage

By default, `Skeleton` will be set to 100% width with a line height of 18px.

```jsx
<Skeleton />
```

### Controlling Size

The height and width of the `Skeleton` can be customized to match the line height of the text that is being loaded. The `size` prop adjusts the line height, and the `characters` prop will adjust the width in increments of the selected `size`.

| `size` Option | Line Height |
| ------------- | ----------- |
| xs            | 18px        |
| sm            | 24px        |
| md            | 36px        |
| lg            | 48px        |
| xl            | 64px        |

```jsx
<Skeleton characters={10} size="md" />
```
