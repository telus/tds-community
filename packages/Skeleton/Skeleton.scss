@import '~@tds/core-colours/colours';

.skeleton-text {
  display: inline-block;
  overflow: hidden;
  max-width: 100%;

  background-color: $color-gainsboro;
  background-repeat: no-repeat;
  background-position: 0 0;
  background-size: 200% 100%;
  background-image: linear-gradient(
    to right,
    $color-gainsboro,
    $color-athens-grey,
    $color-gainsboro
  );
  animation-name: skeleton-text-shimmer;
  animation-duration: 2s;
  animation-delay: 2s;
  animation-iteration-count: infinite;
}

@keyframes skeleton-text-shimmer {
  0% {
    background-position: 200% 0;
  }
  100% {
    background-position: -100% 0;
  }
}

.size-18 {
  height: 18px;
  min-width: 18px;
  border-radius: 9px;
}

.size-24 {
  height: 24px;
  min-width: 24px;
  border-radius: 12px;
}

.size-36 {
  height: 36px;
  min-width: 36px;
  border-radius: 18px;
}

.size-48 {
  height: 48px;
  min-width: 48px;
  border-radius: 24px;
}

.size-64 {
  height: 64px;
  min-width: 64px;
  border-radius: 32px;
}
