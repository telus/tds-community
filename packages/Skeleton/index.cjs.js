const Skeleton = require('./dist/index.cjs')

module.exports = Skeleton
