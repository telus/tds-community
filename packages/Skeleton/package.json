{
  "name": "@tds/community-skeleton",
  "version": "0.1.0",
  "description": "",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds-community.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds-community/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  },
  "dependencies": {
    "@tds/core-colours": "^1.0.0",
    "prop-types": "^15.6.0"
  }
}
