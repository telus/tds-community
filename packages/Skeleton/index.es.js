import Skeleton from './dist/index.es'

export default Skeleton
