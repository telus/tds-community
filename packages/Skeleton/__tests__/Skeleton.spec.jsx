import React from 'react'
import { shallow, mount } from 'enzyme'

import Skeleton, { SIZES } from '../Skeleton'

const doShallow = (props = {}) => shallow(<Skeleton {...props} />)
const doMount = (props = {}) => mount(<Skeleton {...props} />)

describe('Skeleton', () => {
  it('renders', () => {
    const skeleton = doShallow()

    expect(skeleton).toExist()
    expect(skeleton).toMatchSnapshot()
  })

  it('should set correct class based on size attribute, for all sizes', () => {
    const sizes = Object.keys(SIZES)

    expect.assertions(sizes.length)
    sizes.forEach(size => {
      const skeleton = doMount({ size })

      expect(skeleton).toMatchSnapshot()
    })
  })

  it('passes additional attributes to the element', () => {
    const skeleton = doShallow({
      id: 'the-id-but-not-in-the-freudian-sense',
      'data-test': 'automated-e2e-ftw',
    })

    expect(skeleton).toHaveProp('id', 'the-id-but-not-in-the-freudian-sense')
    expect(skeleton).toHaveProp('data-test', 'automated-e2e-ftw')
  })

  it('does not allow custom CSS', () => {
    const skeleton = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(skeleton).not.toHaveProp('className', 'my-custom-class')
    expect(skeleton).not.toHaveProp('style')
  })
})
