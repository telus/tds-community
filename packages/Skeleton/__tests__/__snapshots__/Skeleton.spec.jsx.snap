// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton renders 1`] = `
<Skeleton__StyledSkeleton
  aria-busy="true"
  characters={0}
  isFixedCharacterWidth={false}
  safeSize={18}
/>
`;

exports[`Skeleton should set correct class based on size attribute, for all sizes 1`] = `
.c0 {
  height: 18px;
  min-width: 18px;
  border-radius: 9px;
  width: 100%;
  max-width: 100%;
  display: inline-block;
  overflow: hidden;
  background-color: #d8d8d8;
  background-repeat: no-repeat;
  background-position: 0 0;
  background-size: 200% 100%;
  background-image: linear-gradient( to right,#d8d8d8,#f7f7f8,#d8d8d8 );
  -webkit-animation-name: gaOlHX;
  animation-name: gaOlHX;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-delay: 2s;
  animation-delay: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<Skeleton
  characters={0}
  size="xs"
>
  <Skeleton__StyledSkeleton
    aria-busy="true"
    characters={0}
    isFixedCharacterWidth={false}
    safeSize={18}
  >
    <StyledComponent
      aria-busy="true"
      characters={0}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
              " width:",
              [Function],
              ";max-width:100%;display:inline-block;overflow:hidden;background-color:",
              "#d8d8d8",
              ";background-repeat:no-repeat;background-position:0 0;background-size:200% 100%;background-image:linear-gradient( to right,",
              "#d8d8d8",
              ",",
              "#f7f7f8",
              ",",
              "#d8d8d8",
              " );animation-name:",
              Keyframes {
                "id": "sc-keyframes-gaOlHX",
                "inject": [Function],
                "name": "gaOlHX",
                "rules": Array [
                  "@-webkit-keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                  "@keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                ],
                "toString": [Function],
              },
              ";animation-duration:2s;animation-delay:2s;animation-iteration-count:infinite;",
            ],
          },
          "displayName": "Skeleton__StyledSkeleton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isFixedCharacterWidth={false}
      safeSize={18}
    >
      <span
        aria-busy="true"
        className="c0"
      />
    </StyledComponent>
  </Skeleton__StyledSkeleton>
</Skeleton>
`;

exports[`Skeleton should set correct class based on size attribute, for all sizes 2`] = `
.c0 {
  height: 24px;
  min-width: 24px;
  border-radius: 12px;
  width: 100%;
  max-width: 100%;
  display: inline-block;
  overflow: hidden;
  background-color: #d8d8d8;
  background-repeat: no-repeat;
  background-position: 0 0;
  background-size: 200% 100%;
  background-image: linear-gradient( to right,#d8d8d8,#f7f7f8,#d8d8d8 );
  -webkit-animation-name: gaOlHX;
  animation-name: gaOlHX;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-delay: 2s;
  animation-delay: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<Skeleton
  characters={0}
  size="sm"
>
  <Skeleton__StyledSkeleton
    aria-busy="true"
    characters={0}
    isFixedCharacterWidth={false}
    safeSize={24}
  >
    <StyledComponent
      aria-busy="true"
      characters={0}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
              " width:",
              [Function],
              ";max-width:100%;display:inline-block;overflow:hidden;background-color:",
              "#d8d8d8",
              ";background-repeat:no-repeat;background-position:0 0;background-size:200% 100%;background-image:linear-gradient( to right,",
              "#d8d8d8",
              ",",
              "#f7f7f8",
              ",",
              "#d8d8d8",
              " );animation-name:",
              Keyframes {
                "id": "sc-keyframes-gaOlHX",
                "inject": [Function],
                "name": "gaOlHX",
                "rules": Array [
                  "@-webkit-keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                  "@keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                ],
                "toString": [Function],
              },
              ";animation-duration:2s;animation-delay:2s;animation-iteration-count:infinite;",
            ],
          },
          "displayName": "Skeleton__StyledSkeleton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isFixedCharacterWidth={false}
      safeSize={24}
    >
      <span
        aria-busy="true"
        className="c0"
      />
    </StyledComponent>
  </Skeleton__StyledSkeleton>
</Skeleton>
`;

exports[`Skeleton should set correct class based on size attribute, for all sizes 3`] = `
.c0 {
  height: 36px;
  min-width: 36px;
  border-radius: 18px;
  width: 100%;
  max-width: 100%;
  display: inline-block;
  overflow: hidden;
  background-color: #d8d8d8;
  background-repeat: no-repeat;
  background-position: 0 0;
  background-size: 200% 100%;
  background-image: linear-gradient( to right,#d8d8d8,#f7f7f8,#d8d8d8 );
  -webkit-animation-name: gaOlHX;
  animation-name: gaOlHX;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-delay: 2s;
  animation-delay: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<Skeleton
  characters={0}
  size="md"
>
  <Skeleton__StyledSkeleton
    aria-busy="true"
    characters={0}
    isFixedCharacterWidth={false}
    safeSize={36}
  >
    <StyledComponent
      aria-busy="true"
      characters={0}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
              " width:",
              [Function],
              ";max-width:100%;display:inline-block;overflow:hidden;background-color:",
              "#d8d8d8",
              ";background-repeat:no-repeat;background-position:0 0;background-size:200% 100%;background-image:linear-gradient( to right,",
              "#d8d8d8",
              ",",
              "#f7f7f8",
              ",",
              "#d8d8d8",
              " );animation-name:",
              Keyframes {
                "id": "sc-keyframes-gaOlHX",
                "inject": [Function],
                "name": "gaOlHX",
                "rules": Array [
                  "@-webkit-keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                  "@keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                ],
                "toString": [Function],
              },
              ";animation-duration:2s;animation-delay:2s;animation-iteration-count:infinite;",
            ],
          },
          "displayName": "Skeleton__StyledSkeleton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isFixedCharacterWidth={false}
      safeSize={36}
    >
      <span
        aria-busy="true"
        className="c0"
      />
    </StyledComponent>
  </Skeleton__StyledSkeleton>
</Skeleton>
`;

exports[`Skeleton should set correct class based on size attribute, for all sizes 4`] = `
.c0 {
  height: 48px;
  min-width: 48px;
  border-radius: 24px;
  width: 100%;
  max-width: 100%;
  display: inline-block;
  overflow: hidden;
  background-color: #d8d8d8;
  background-repeat: no-repeat;
  background-position: 0 0;
  background-size: 200% 100%;
  background-image: linear-gradient( to right,#d8d8d8,#f7f7f8,#d8d8d8 );
  -webkit-animation-name: gaOlHX;
  animation-name: gaOlHX;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-delay: 2s;
  animation-delay: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<Skeleton
  characters={0}
  size="lg"
>
  <Skeleton__StyledSkeleton
    aria-busy="true"
    characters={0}
    isFixedCharacterWidth={false}
    safeSize={48}
  >
    <StyledComponent
      aria-busy="true"
      characters={0}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
              " width:",
              [Function],
              ";max-width:100%;display:inline-block;overflow:hidden;background-color:",
              "#d8d8d8",
              ";background-repeat:no-repeat;background-position:0 0;background-size:200% 100%;background-image:linear-gradient( to right,",
              "#d8d8d8",
              ",",
              "#f7f7f8",
              ",",
              "#d8d8d8",
              " );animation-name:",
              Keyframes {
                "id": "sc-keyframes-gaOlHX",
                "inject": [Function],
                "name": "gaOlHX",
                "rules": Array [
                  "@-webkit-keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                  "@keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                ],
                "toString": [Function],
              },
              ";animation-duration:2s;animation-delay:2s;animation-iteration-count:infinite;",
            ],
          },
          "displayName": "Skeleton__StyledSkeleton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isFixedCharacterWidth={false}
      safeSize={48}
    >
      <span
        aria-busy="true"
        className="c0"
      />
    </StyledComponent>
  </Skeleton__StyledSkeleton>
</Skeleton>
`;

exports[`Skeleton should set correct class based on size attribute, for all sizes 5`] = `
.c0 {
  height: 64px;
  min-width: 64px;
  border-radius: 32px;
  width: 100%;
  max-width: 100%;
  display: inline-block;
  overflow: hidden;
  background-color: #d8d8d8;
  background-repeat: no-repeat;
  background-position: 0 0;
  background-size: 200% 100%;
  background-image: linear-gradient( to right,#d8d8d8,#f7f7f8,#d8d8d8 );
  -webkit-animation-name: gaOlHX;
  animation-name: gaOlHX;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-delay: 2s;
  animation-delay: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<Skeleton
  characters={0}
  size="xl"
>
  <Skeleton__StyledSkeleton
    aria-busy="true"
    characters={0}
    isFixedCharacterWidth={false}
    safeSize={64}
  >
    <StyledComponent
      aria-busy="true"
      characters={0}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              [Function],
              " width:",
              [Function],
              ";max-width:100%;display:inline-block;overflow:hidden;background-color:",
              "#d8d8d8",
              ";background-repeat:no-repeat;background-position:0 0;background-size:200% 100%;background-image:linear-gradient( to right,",
              "#d8d8d8",
              ",",
              "#f7f7f8",
              ",",
              "#d8d8d8",
              " );animation-name:",
              Keyframes {
                "id": "sc-keyframes-gaOlHX",
                "inject": [Function],
                "name": "gaOlHX",
                "rules": Array [
                  "@-webkit-keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                  "@keyframes gaOlHX{0%{background-position:200% 0;}100%{background-position:-100% 0;}}",
                ],
                "toString": [Function],
              },
              ";animation-duration:2s;animation-delay:2s;animation-iteration-count:infinite;",
            ],
          },
          "displayName": "Skeleton__StyledSkeleton",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Skeleton__StyledSkeleton-sc-4tyjdz-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      isFixedCharacterWidth={false}
      safeSize={64}
    >
      <span
        aria-busy="true"
        className="c0"
      />
    </StyledComponent>
  </Skeleton__StyledSkeleton>
</Skeleton>
`;
